From 9311d491def907ee24f4402bb2ccbffcae6e0f70 Mon Sep 17 00:00:00 2001
From: Mandy Lavi <mandy.lavi@freescale.com>
Date: Sun, 12 May 2013 12:16:34 +0300
Subject: [PATCH 219/547] dpaa_eth: change init sequence so that TX port is
 before RX port

Signed-off-by: Mandy Lavi <mandy.lavi@freescale.com>
VSP init sequence expects TX port to be initialized prior RX port
Change-Id: I72f0e35d935de15647fa28a975cd89f04151fefd
Reviewed-on: http://git.am.freescale.net:8181/2448
Reviewed-by: Hamciuc Bogdan-BHAMCIU1 <bogdan.hamciuc@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
[Original patch taken from QorIQ-SDK-V1.4-SOURCE-20130625-yocto.iso]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
index 3a07949..e56a894 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
@@ -4180,10 +4180,10 @@ dpaa_eth_probe(struct platform_device *_of_dev)
 	if (mac_dev) {
 		struct fm_port_pcd_param rx_port_pcd_param;
 
-		dpaa_eth_init_rx_port(rxport, dpa_bp, count, rxerror,
-				      rxdefault, &buf_layout[RX]);
 		dpaa_eth_init_tx_port(txport, txerror, txdefault,
 				      &buf_layout[TX]);
+		dpaa_eth_init_rx_port(rxport, dpa_bp, count, rxerror,
+				      rxdefault, &buf_layout[RX]);
 
 		rx_port_pcd_param.cba = dpa_alloc_pcd_fqids;
 		rx_port_pcd_param.cbf = dpa_free_pcd_fqids;
-- 
1.8.4.93.g57e4c17

